connectors/sources/tests/test_confluence.py

```diff
@@ -157,16 +162,89 @@ async def test_validate_configuration_with_invalid_concurrent_downloads():
     """Test validate configuration method of BaseDataSource class with invalid concurrent downloads"""
 
     # Setup
-    source = create_source(ConfluenceDataSource)
-    source.concurrent_downloads = 1000
+    source = create_source(ConfluenceDataSource, concurrent_downloads=1000)
 
     # Execute
-    with pytest.raises(
-        Exception, match="Configured concurrent downloads can't be set more than *"
-    ):
+    with pytest.raises(ConfigurableFieldValueError):
         await source.validate_config()
 
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "extras",
+    [
+        (
+            # Confluence Server with blank dependent fields
+            {
+                "data_source": CONFLUENCE_SERVER,
+                "username": "",
+                "password": "",
+                "account_email": "foo@bar.me",
+                "api_token": "foo",
+            }
+        ),
+        (
+            # Confluence Cloud with blank dependent fields
+            {
+                "data_source": CONFLUENCE_CLOUD,
+                "username": "foo",
+                "password": "bar",
+                "account_email": "",
+                "api_token": "",
+            }
+        ),
+    ],
+)
+async def test_validate_configuration_with_invalid_dependency_fields_raises_error(
+    extras,
+):
+    # Setup
+    source = create_source(ConfluenceDataSource, **extras)
+
+    # Execute
+    with pytest.raises(ConfigurableFieldValueError):
+        await source.validate_config()
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "extras",
+    [
+        (
+            # Confluence Server with blank non-dependent fields
+            {
+                "data_source": CONFLUENCE_SERVER,
+                "username": "foo",
+                "password": "bar",
+                "account_email": "",
+                "api_token": "",
+            }
+        ),
+        (
+            # Confluence Cloud with blank non-dependent fields
+            {
+                "data_source": CONFLUENCE_CLOUD,
+                "username": "",
+                "password": "",
+                "account_email": "foo@bar.me",
+                "api_token": "foobar",
+            }
+        ),
+        (
+            # SSL certificate not enabled (empty ssl_ca okay)
+            {"ssl_enabled": False, "ssl_ca": ""}
+        ),
+        # TODO: add test case with a mocked SSL cert
```

Comment:
Note: I couldn't figure out how to do this, hence the TODO

Reply:
@timgrein so I tried again, and couldn't get it to work. I decided to make a dedicated test for it so I could patch `ssl_context` only when needed.
New test is below, named `test_validate_config_with_valid_dependency_fields_does_not_raise_error`